pkg/kubelet/stats/cadvisor_stats_provider.go

```diff
@@ -192,6 +204,28 @@ func isPodManagedContainer(cinfo *cadvisorapiv2.ContainerInfo) bool {
 	return managed
 }
 
+// getcadvisorPodInfoFromPodUID returns a pod cgroup information based on the podUID
+// In ContainerInfo data struct, the key for pod cgroup is normally one of the following format (<podUID> is the string of the pod UID)
```

Comment:
> ... the key for pod cgroup is normally one of the following ...

When would it not be one of the following?

Reply:
Actually, it feels like the cgroups manager should be able to just tell us what the pod crgoup path should be? Then you can do a full path match.